Signal/src/call/CallAudioService.swift

```diff
@@ -178,16 +180,22 @@ struct AudioSource: Hashable {
             setAudioSession(category: AVAudioSessionCategoryPlayAndRecord,
                             mode: AVAudioSessionModeVideoChat,
                             options: options)
-        } else if call.isSpeakerphoneEnabled {
-            // Ensure no bluetooth if user has specified speakerphone
-            setAudioSession(category: AVAudioSessionCategoryPlayAndRecord,
-                            mode: AVAudioSessionModeVoiceChat,
-                            options: [.defaultToSpeaker])
         } else {
             setAudioSession(category: AVAudioSessionCategoryPlayAndRecord,
                             mode: AVAudioSessionModeVoiceChat,
                             options: [.allowBluetooth])
         }
+
+        let session = AVAudioSession.sharedInstance()
+        do {
+            if call.isSpeakerphoneEnabled {
+                try session.overrideOutputAudioPort(.speaker)
```

Comment:
Is this cheap?  `setAudioSession` ignored redundant changes - can/should we do something like that here?